spec/template.dd

```diff
@@ -36,11 +38,9 @@ $(GNAME TemplateParameter):
     )
 
     $(P Template parameters can be types, values, symbols, or sequences.
-        Types can be any type.
-        Value parameters must be of an integral type, floating point
-        type, or string type and
-        specializations for them must resolve to an integral constant,
-        floating point constant, null, or a string literal.
+        Type parameters can be any type.
+        Value parameters can have any type which can be statically
```

Comment:
Isn't it the case that the *type* of a value parameter can be any valid D type, but whether an argument can be passed to it is the point being made about it being computable at compile?

Reply:
The *argument* can be of any compile-time constructable type. I essentially copied the first sentence of:
https://dlang.org/spec/template.html#template_value_parameter

I would prefer not to repeat the kinds of types which satisfy that requirement, as they are listed in the value parameter section. (There is a link to that just above this paragraph).

Reply:
@maxhaton I've now tweaked this to say *Value parameters can take any value which can be statically initialized at compile time*, so it no longer talks about types.

Reply:
Another tweak - *can take any expression...*. Also I restored the sentence about the specialization expression needing to be CT evaluatable, but I put it in the *TemplateValueParameter* section instead as that seems more appropriate - 4a0300c.